pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +92,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthPlugin is a custom plugin used to authenticate to the kubernetes cluster.
+	AuthPlugin string `json:"auth-plugin,omitempty"`
```

Comment:
ConfigMap would be less flexible for plugin builders (string->string map instead of generic JSON blob) without providing much of a benefit.

I need to look into whether it'd be better to use something like `runtime.RawExtension` & `runtime.Object`. I'm hoping @smarterclayton or @krousey can kick in some feedback on that.